### ISSUE TITLE
Initilization error and VTK outputs prevents a clean abort

### ISSUE THREAD

**Issue by ebranlard** [OpenFAST/openfast#250]
**Bug description**

If an error occurs during the fast library initialization, the routine `ExitThisProgram` is called, which then attempts to write the mesh for debugging if `WrVTK>0`. The issue is that  `y_FAST%NOutSteps` might be zero, and then the program will crash on the following statement `Twidth = int(log10(real(y_FAST%NOutSteps))) + 1` . 

**To Reproduce**

Try to trigger an error at initialization (e.g. by reproducing #249), and set `WrVTK` to on. 

**Expected behavior**

The program should abort nicely. 

**Suggested solution**

The calls to `log10` in `FAST_Subs.f90` should be protected as follows: 
```fortran
   if (y_FAST%NOutSteps==0) then
      CALL SetErrStat( ErrID_Warn,'NOutStep is zero, setting Twidth to 1',ErrStat2,ErrMsg2,RoutineName)
      Twidth = 1
   else
       Twidth = int(log10(real(y_FAST%NOutSteps))) + 1
   endif
```
I'll send a pull request later that fixes this if people agree.

**Comment by bjonkman** [comment]
I think that looks good. I'm not sure the warning is necessary, though.

**Comment by ebranlard** [comment]
The pullrequest #257 solves this issue but I have now discovered a related bug.  `NOutStep` is used for `Twidth`, but `NOutStep` is only defined when `WrBinOutFile` is on. I also believe that `NOutStep` should be different for VTK files, based on the `n_VTKTime` for instance. I'll investigate a bit more.

**Comment by ebranlard** [comment]
I'm suggesting the following changes in `FAST_Subs.f90`, routine `FAST_InitOutput`:

- Always compute `NOutSteps`, not only when `WrBinOutFile` is true, OR, initialize it to 1. It needs initialization since gfortran for instance will not initialize it to 0.

- After `NOutSteps` is set, add the following, to take the maximum between this value and the number of outputs required by the `VTK_fps` parameter. 
```fortran
   IF (p_FAST%n_VTKTime>0) THEN
       DT_OUT_VTK = p_FAST%n_VTKTime * p_FAST%DT
       y_FAST%NOutSteps = MAX ( y_FAST%NOutSteps, CEILING ( (p_FAST%TMax - p_FAST%TStart) / DT_OUT_VTK ) + 1 )
   ENDIF
```

**Comment by bjonkman** [comment]
I don't think `NOutSteps` should be used in the calculation for `tWidth` at all... The VTK files are written at a rate independent from the rate the normal FAST output files are written, and the VTK output start at t=0, not TStart.

In my version of the code, I use 
```
tWidth=CEILING( log10( real(p_FAST%n_TMax_m1+1, ReKi) / p_FAST%n_VTKTime ) ) + 1
``` 
This is actually what I had suggested in the first issue reported in https://github.com/OpenFAST/openfast/issues/172, and which I thought had already been fixed.

**Comment by ebranlard** [comment]
Thanks for pointing out #172, it is indeed related to this second issue I mentioned. Unfortunately, it seems the change you suggested are not included in the current dev branch. 

I agree that the rates are different, my solution was more of a quick fix. Probably the best would be to pass twidth as an argument to the write functions, but as long as these functions are only used for the visualization, it seems fine. 

I can submit a pull request that include your version of `twidth`, wrapping it with an if statement to avoid any division by zero or log10 of 0.

**Comment by rafmudaf** [comment]
closed by pull request #257
